### PR TITLE
fix logo not correctly inverted

### LINKED ISSUE #34656: [Bug]: Background pictures which invert the heading icons do not invert the logo
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When picking a light background, the navigation icons invert but the logo stays white, looking inconsistent.

image

### Steps to reproduce

1. Go to "Appearance & accessibility"
2. Pick a light background
3. See how the app icons invert, but the logo not

### Expected behavior

If the logo is still our default Nextcloud one, we could also invert it as it looks much nicer. Themed logos should of course not be inverted.

image

### Installation method

_No response_

### Operating system

_No response_

### PHP engine version

_No response_

### Web server

_No response_

### Database engine version

_No response_

### Is this bug present after an update or on a fresh install?

_No response_

### Are you using the Nextcloud Server Encryption module?

_No response_

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
.
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/server/issues/34656

| Before | After |
|---|---|
| image | image |

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
@nextcloud/server-frontend I suppose we only want this to happen if the default logo is used. Can we somehow implement a check for this? I failed to find a solution for this...

**Comment by CarlSchwan** [comment]
The proper solution would be to allow admin to provide a dark and a white themed logo

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 1]
```diff
@@ -203,6 +203,8 @@ public function getCSSVariables(): array {
 			'--background-invert-if-bright' => 'invert(100%)',
 
 			'--image-main-background' => "url('" . $this->urlGenerator->imagePath('core', 'app-background.jpg') . "')",
+
+			'--logo-invert-if-default' => ($hasCustomLogoHeader && $this->util->invertTextColor($this->primaryColor)) ? 'no' : 'invert(100%)',
```
Comment (szaimen) [inline apps/theming/lib/Themes/DefaultTheme.php:207]: seems like it doesnt work, yet

**Comment by szaimen** [comment]
Seems like we need to fix https://github.com/nextcloud/server/issues/33969 first...

**Comment by jancborchardt** [comment]
> I suppose we only want this to happen if the default logo is used.

@szaimen Yes, as already said in https://github.com/nextcloud/server/issues/33969 :)

> we need to invert the logo as well (only if it is our default Nextcloud logo though)

**Comment by szaimen** [comment]
@skjnldsv an you take this over? I am currently a bit out of ideas how to solve this. Otherwise I would appreciate some pointers on this :)

**Comment by skjnldsv** [comment]
Done, please check @szaimen

**Review by skjnldsv — APPROVED** [review]

File: apps/theming/lib/Util.php [hunk 2]
```diff
@@ -266,4 +261,9 @@ public function isBackgroundThemed() {
 		$backgroundLogo = $this->config->getAppValue('theming', 'backgroundMime', '');
 		return $backgroundLogo !== '' && $backgroundLogo !== 'backgroundColor';
 	}
+
+	public function isLogoThemed() {
```
Comment (CarlSchwan) [inline apps/theming/lib/Util.php:265]:
```suggestion
	public function isLogoThemed(): bool {
```

Comment (skjnldsv) [reply apps/theming/lib/Util.php:265]: Just let get this in please ... :cry:

**Review by szaimen — COMMENTED** [review]
This doesnt seem to work correctly yet.
Just tested and it looks like this now with dark mode enabled:
image

**Comment by szaimen** [comment]
@skjnldsv sorry for bothering you but could you maybe take another look at this? I think we need to get the value from the our default themes by checking against its dark value...

**Comment by skjnldsv** [comment]
> @skjnldsv sorry for bothering you but could you maybe take another look at this? I think we need to get the value from the our default themes by checking against its dark value...

My bad, wrong fallback value! :+1: 
Fixed!

**Review by szaimen — COMMENTED** [review]
Seems to work correclty now in my testing so ready to go from my side! :)
Unfortunately I cannot approve as it is my PR but you could approve for me :P

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**